### ISSUE TITLE
ENH: Option to configure line wrapping for column names

### ISSUE THREAD

**Issue by Nagidrop** [pandas-dev/pandas#59564]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

When having large enough `display.max_colwidth` value and a column contains cells with long enough data (string, list..), other columns with only short data will get their name (containing whitespaces) line-wrapped, even though there is plenty of space left and there is also a scrollbar to accommodate the large DataFrame width.

This can be reproduced using the snippet below:
```
import pandas as pd

pd.set_option('display.max_colwidth', None)  # None or a large enough value
df = pd.DataFrame({
    'column name': ['data0', 'data1'],
    'col2': ['0', 'a really longggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg string']
})
display(df)
```
![image](https://github.com/user-attachments/assets/ed0d4d69-af72-46c9-b839-181b527fa84b)

I want to make the `column name` shown on 1 line, instead of being line-wrapped to 2 lines. There isn't limited space here so I think linewrapping is unnecessary. I have searched in [Pandas "Options and Settings" page](https://pandas.pydata.org/docs/user_guide/options.html#available-options)) and there is currently no option to configure this behaviour. I would be really grateful if this gets implemented.

### Feature Description

Add a new option to configure linewrapping behaviour for columns, e.g. `display.linewrap` that accepts boolean values, `True` to enable linewrapping and `False` otherwise.

### Alternative Solutions

None

**Comment by rhshadrach** [comment]
Thanks for the request. Why does the line wrapping cause an issue for you?

**Comment by Nagidrop** [comment]
It could be my personal preference only, but I found reading things on 1 line is easier on my eyes. I have crafted an example for you to better illustrate. Here is some data when I use `display.max_colwidth = 38`:

![image](https://github.com/user-attachments/assets/6883967f-5997-47ef-ab7a-ace9925e8f88)

Everything shows up nicely. But there is one review where it's very long (hence ellipsis) and I need to change `display.max_colwidth = None` to show all of its contents:

![image](https://github.com/user-attachments/assets/4206809c-c073-4838-aea0-16c5d861341a)

Now all of its contents are shown, but suddenly everything else is squished together and makes it hard to read, even though there is plenty of room left. The linewrapping behaviour here is unexpected because I only need the rightmost column to expand and show the long review. Also, when you specify a very large column width, logically it should show the column contents on 1 line, not condense and linewrap it (which is only appropriate if column width is small). That's why I think there should be an option to control this behaviour.

**Comment by rhshadrach** [comment]
> makes it hard to read

I can see the preference for not having line breaks, and could buy that with line breaks is harder, but I do not think this is hard to read. At a glance, it seems to me the additional complexity in output formatting is not worth the gain here.

If it could be shown that this change can be made with negligible additional complexity to the current set of features, I could be swayed otherwise.

**Comment by Nagidrop** [comment]
Yeah the automatic linewrapping is generally a non-issue for me. Only in some occasions where it brings unexpected changes, I do feel like we could have some more control there. But if the option addition complicated the code base, I also agree to not implement at all. 

Hopefully I could find a simple solution for this when I have some spare time.
